docs/devel/releasing.md

```diff
@@ -1,7 +1,155 @@
 # Releasing Kubernetes
 
-This document explains how to create a Kubernetes release (as in version) and
-how the version information gets embedded into the built binaries.
+This document explains how to cut a release, and the theory behind it. If you
+just want to cut a release and move on with your life, you can stop reading
+after the first section.
+
+## How to cut a Kubernetes release
+
+Regardless of whether you are cutting a major or minor version, cutting a
+release breaks down into four pieces:
+
+1. Selecting release components.
+1. Tagging and merging the release in Git.
+1. Building and pushing the binaries.
+1. Writing release notes.
+
+You should progress in this strict order without interleaving pieces of each
+major step unless you are very familiar with the release process.
+
+### Building a New Major/Minor Version (`vx.y.0`)
```

Comment:
I've got all sorts of minor nits i'll keep to mymeself, but I do like capital X and Y, to make it easier to distinguish the v...